### PR TITLE
Fix p:f:T::load()

### PR DESCRIPTION
**Pull request by peterrum**

### CODE REVIEW

File: source/distributed/tria_base.cc [hunk 1]
```diff
@@ -684,9 +684,11 @@ namespace parallel
 
   template <int dim, int spacedim>
   DEAL_II_CXX20_REQUIRES((concepts::is_valid_dim_spacedim<dim, spacedim>))
-  void DistributedTriangulationBase<dim, spacedim>::clear()
+  void TriangulationBase<dim, spacedim>::clear()
   {
     dealii::Triangulation<dim, spacedim>::clear();
+
+    number_cache = {};
```
Comment (peterrum) [inline source/distributed/tria_base.cc:691]: The old test passed because the cache was not cleared^^

Comment (blaisb) [reply source/distributed/tria_base.cc:691]: Thank you, I was questioning my sanity. The load feature does not work then for fully distributed triangulations?

**Review by blaisb — APPROVED** [review]

**Comment by blaisb** [comment]
@peterrum there is a paralllel::shared triangulation test that is breaking. Do you know if this is related? I have a feeling no because I tested this PR on my machine on the bug I had.

File: include/deal.II/distributed/shared_tria.h [hunk 2]
```diff
@@ -319,6 +319,21 @@ namespace parallel
...
+       * of serialization. using the [BOOST serialization
+       * library](https://www.boost.org/doc/libs/1_74_0/libs/serialization/doc/index.html).
+       */
+      template <class Archive>
+      void
+      serialize(Archive &archive, const unsigned int version);
+#  else
+      // This macro defines the serialize() method that is compatible with
+      // the templated save() and load() method that have been implemented.
+      BOOST_SERIALIZATION_SPLIT_MEMBER()
```
Comment (peterrum) [inline include/deal.II/distributed/shared_tria.h:334]: Without this, boost did not pick up the specializations of `load`/`save` of p:s:T...

File: include/deal.II/distributed/tria_base.h [hunk 3]
```diff
@@ -304,6 +304,15 @@ namespace parallel
     virtual types::coarse_cell_id
     n_global_coarse_cells() const override;
 
+    /**
+     * Reset this triangulation into a virgin state by deleting all data.
```
Comment (bangerth) [inline include/deal.II/distributed/tria_base.h:308]: Let's avoid the void "virgin". Can you change it to "to an empty state"? I will update all of the other places in the library that use the word.

Comment (blaisb) [reply include/deal.II/distributed/tria_base.h:308]: Ahh you know what, that's such a good idea. Reminds me of the main/master thing.
Small changes that make no real difference, but makes everyone safer and happier.

Comment (peterrum) [reply include/deal.II/distributed/tria_base.h]: Done :+1:

**Comment by blaisb** [comment]
The test still seem to break @peterrum  :( Unsure what's going on here.

**Comment by peterrum** [comment]
> The test still seem to break @peterrum :( Unsure what's going on here.

@blaisb  The test output was not updated.

**Comment by blaisb** [comment]
> > The test still seem to break @peterrum :( Unsure what's going on here.
> 
> @blaisb The test output was not updated.

Perfect
I'll merge after CI then :)
Thanks for taking care of this. Really appreciate it :)

**Comment by blaisb** [comment]
if everybody is ok with this I would proceed with a merge?
@peterrum @bangerth ?

**Pull request merged**